rat/model.py

```diff
@@ -25,7 +27,32 @@ class Model:
     parameter_offsets: List[int]
     parameter_sizes: List[Union[None, int]]
 
-    def __init__(self, data_df: pandas.DataFrame, parsed_lines: List[ops.Expr]):
+    def __init__(
+        self,
+        data_df: pandas.DataFrame,
+        parsed_lines: List[ops.Expr] = None,
+        model_string: str = None,
+    ):
+        data_names = data_df.columns
+        if model_string is not None:
+            if parsed_lines is not None:
+                raise Exception(
+                    "Only one of model_string and parsed_lines can be non-None"
+                )
+
+            parsed_lines = []
+            for line in model_string.splitlines():
```

Comment:
One thing I might try is just wrap the entire multiline parsing into a function of the parser. This might be useful in the future in cases something changes and the spllit() works on different separators.